third_party/toolchains/backwards_compatibility.bzl

```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@bazel_toolchains//rules/exec_properties:exec_properties.bzl", "create_exec_properties_dict", "create_rbe_exec_properties_dict","merge_dicts")
+
+def bc_exec_properties(exec_properties):
```

Comment:
I think this function deserves a comment. 
Basically, for newer bazel toolchains we should just be invoking `create_rbe_exec_properties_dict(xyz=abc, foo=bar, labels = { ... })` but we cannot do that for backward compatibility. There should be a comment that provides details of what doesn't work on which bazel/toolchain version.


https://github.com/grpc/grpc/pull/25001/files#diff-5493ff8e9397811510e780de47c57abb70137f1afe85d1519130dc3679d60ce5R40